### ISSUE TITLE
Error when starting application

### ISSUE THREAD

**Issue by RogerHaugen** [Micke-K/IntuneManagement#21]
```
Using MSAL file C:\IntuneManagement-master\Microsoft.Identity.Client.dll. Version: 4.29.0.0
Add-Type : Ingen tilgang til banen C:\IntuneManagement-master\CS\TokenCacheHelperEx.cs.
At C:\IntuneManagement-master\Extensions\MSALAuthentication.psm1:407 char:5
+     Add-Type -Path ($global:AppRootFolder + "\CS\TokenCacheHelperEx.c ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Add-Type], UnauthorizedAccessException
    + FullyQualifiedErrorId : System.UnauthorizedAccessException,Microsoft.PowerShell.Commands.AddTypeCommand
```

Translation: Add-Type : Ingen tilgang til banen - No access to the filepath

All files are untouched, and i see dll file in the dir

**Comment by Micke-K** [comment]
This suggests  that the user you started the app with does not have write access to the C:\IntuneManagement-master\CS folder. Can you check the permission on the folder? Or copy the whole IntuneManagement-master to your downloads (or OneDrive) and try from there

**Comment by Micke-K** [comment]
Did you get it to start?

**Comment by VRDRF** [comment]
Seeing the same problem here but I seem to have found the issue, ASLR seems to not like the file:
![image](https://user-images.githubusercontent.com/88138218/127467106-e44df542-daa7-42ce-8001-9163c6a5cdbe.png)

**Comment by Micke-K** [comment]
Not sure if it is the same issue. Yours is clearly an ASR issue and I expect the original error is no write access to the folder. I can reproduce that error. 

I can see why ASR would see the CS file as obfuscated. It is in clear text and 95% based on code from MS. This is used when caching the the token. You could try to disable that in Settings are restart. Does that work?

**Comment by VRDRF** [comment]
Yeah I manage to get it to work by whitelisting the location :)

**Comment by Micke-K** [comment]
There is a fix for this in the 3.2 version. Please let me know if I can close this now.
